backend/remote-state/azure/client.go

```diff
@@ -39,7 +39,7 @@ func (c *RemoteClient) Get() (*remote.Payload, error) {
 	ctx := context.TODO()
 	blob, err := c.giovanniBlobClient.Get(ctx, c.accountName, c.containerName, c.keyName, options)
 	if err != nil {
-		if blob.Response.StatusCode == 404 {
+		if blob.Response.Response != nil && blob.Response.StatusCode == 404 {
```

Comment:
this can become:

```suggestion
		if blob.Response.IsHTTPStatus(http.StatusNotFound) {
```

which should handle this being nil and avoid the crash, since internally this is:

```
func (r Response) IsHTTPStatus(statusCode int) bool {
	if r.Response == nil {
		return false
	}
	return r.Response.StatusCode == statusCode
}
```

As such could we update this?